File: app/src/main/java/com/zingkg/arkhamhorrorassistant/app/fragments/CultEncounterFragment.java
package com.zingkg.arkhamhorrorassistant.app.fragments;

import android.os.Bundle;
import android.text.Html;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.zingkg.arkhamhorrorassistant.app.R;
import com.zingkg.arkhamhorrorassistant.xml.CultEncounter;

public class CultEncounterFragment extends DeckFragment {
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setHasOptionsMenu(true);
    }

    @Override
    public View onCreateView(
        LayoutInflater inflater,
        ViewGroup container,
        Bundle savedInstanceState
    ) {
        super.onCreateView(inflater, container, savedInstanceState);
        return inflater.inflate(R.layout.fragment_cult_encounter, container, false);
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        CultEncounter card = new CultEncounter(
            getArguments().getString("title", ""),
            getArguments().getString("lore", ""),
            getArguments().getString("entry", ""),
            getArguments().getString("expansionSet", "")
        );
        TextView titleView = (TextView) view.findViewById(R.id.cult_encounter_title);
        if (card.lore.isEmpty() && card.entry.isEmpty()) {
            view.findViewById(R.id.cult_encounter_title_divider).setVisibility(View.INVISIBLE);
            titleView.setTextSize(getTitleDimensionPixelSize());
        } else {
            TextView loreText = (TextView) view.findViewById(R.id.cult_encounter_lore);
            TextView entryText = (TextView) view.findViewById(R.id.cult_encounter_entry);
            final float textSize = calculateCardTextSize(getScreenLength());
            loreText.setText(Html.fromHtml(card.lore));
            loreText.setTextSize(textSize);
            entryText.setText(Html.fromHtml(card.entry));
            entryText.setTextSize(textSize);
        }
        titleView.setText(Html.fromHtml(card.title));
    }

    @Override
    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        super.onCreateOptionsMenu(menu, inflater);
        inflater.inflate(R.menu.menu_decks, menu);
    }
}

File: app/src/main/kotlin/com/zingkg/arkhamhorrorboard/secondedition/app/fragments/CultEncounterFragment.kt
package com.zingkg.arkhamhorrorboard.secondedition.app.fragments

import android.os.Bundle
import android.text.Html
import android.view.LayoutInflater
import android.view.Menu
import android.view.MenuInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.zingkg.arkhamhorrorboard.secondedition.app.R
import com.zingkg.arkhamhorrorboard.secondedition.card.Cards
import com.zingkg.arkhamhorrorboard.secondedition.card.CultEncounter

class CultEncounterFragment : CardFragment() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setHasOptionsMenu(true)
  }

  override fun onCreateView(
    inflater: LayoutInflater?,
    container: ViewGroup?,
    savedInstanceState: Bundle?
  ): View? {
    super.onCreateView(inflater, container, savedInstanceState)
    return inflater?.inflate(R.layout.fragment_cult_encounter, container, false)
  }

  override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
    super.onViewCreated(view, savedInstanceState)
    val (title, lore, entry, expansionSet) = importCultEncounter(arguments)
    if (lore.isEmpty() && entry.isEmpty()) {
      view?.findViewById<View>(R.id.cult_encounter_title_divider)?.visibility = View.INVISIBLE
      view?.findViewById<View>(R.id.cult_encounter_expansion_set_divider)?.visibility = View.INVISIBLE
    } else {
      view?.findViewById<TextView>(R.id.cult_encounter_lore)?.text = Html.fromHtml(lore)
      view?.findViewById<TextView>(R.id.cult_encounter_entry)?.text = Html.fromHtml(entry)
      view?.findViewById<TextView>(R.id.cult_encounter_expansion_set)?.text = Html.fromHtml(expansionSet.shortName)
    }
    (view?.findViewById<View>(R.id.cult_encounter_title) as TextView).text = Html.fromHtml(title)
  }

  override fun onCreateOptionsMenu(menu: Menu?, inflater: MenuInflater?) {
    super.onCreateOptionsMenu(menu, inflater)
    inflater?.inflate(R.menu.menu_decks, menu)
  }

  companion object {
    private const val TITLE_KEY = "title"
    private const val LORE_KEY = "lore"
    private const val ENTRY_KEY = "entry"
    private const val EXPANSION_SET_KEY = "expansionSet"

    fun exportCultEncounter(cultEncounter: CultEncounter): Bundle {
      val bundle = Bundle()
      bundle.putString(TITLE_KEY, cultEncounter.title)
      bundle.putString(LORE_KEY, cultEncounter.lore)
      bundle.putString(ENTRY_KEY, cultEncounter.entry)
      bundle.putString(EXPANSION_SET_KEY, cultEncounter.expansionSet.toString())
      return bundle
    }

    fun importCultEncounter(bundle: Bundle): CultEncounter {
      return CultEncounter(
        bundle.getString(TITLE_KEY, ""),
        bundle.getString(LORE_KEY, ""),
        bundle.getString(ENTRY_KEY, ""),
        Cards.ExpansionSet.valueOf(bundle.getString(EXPANSION_SET_KEY, Cards.ExpansionSet.BASE.name))
      )
    }
  }
}
